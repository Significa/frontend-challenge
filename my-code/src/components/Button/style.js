import sc, { css } from 'styled-components'
import { theme } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.button`
  display: flex;
  align-items: center;
  /* Account for border */
  padding: 0 15px 0 11px;
  height: 40px;
  outline: none;
  background: none;
  color: ${theme('greyLight')};
  border: 1px solid ${theme('greyMid')};
  border-radius: 4px;
  cursor: pointer;
  transition-timing-function: ${theme('transitionTiming')};
  transition-duration: ${theme('transitionDuration')};
  transition-property: color, border-color, background-color;
  font-family: inherit;
  ${text4}
  font-weight: 500;

  &:focus,
  &:hover {
    border-color: ${theme('mainRed')};
    color: ${theme('greyWhite')};
  }

  ${({ active }) =>
    active &&
    css`
      border-color: ${theme('mainRed')};
      background-color: ${theme('mainRed')};
      color: ${theme('greyWhite')};
    `}
`

export const IconWrapper = sc.div`
  width: 16px;
  height: 16px;
  margin-right: 12px;
  display: flex;
  align-items: center;
`
