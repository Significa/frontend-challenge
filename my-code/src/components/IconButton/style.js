import sc from 'styled-components'
import { theme } from 'styled-tools'

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = sc.button`
  height: 40px;
  outline: none;
  border: none;
  padding: 0;
  background: none;
  color: ${theme('greyLight')};
  cursor: pointer;
  transition:
    color
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;
  display: flex;
  align-items: center;
  justify-content: center;

  &:focus,
  &:hover {
    color: ${theme('greyWhite')};
  }
`
