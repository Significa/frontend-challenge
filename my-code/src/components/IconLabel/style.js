import sc from 'styled-components'
import { theme, prop } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.div`
  display: flex;
  align-items: stretch;
  height: 40px;
  margin-right: 16px;
  margin-bottom: 8px;
`

export const IconWrapper = sc.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${prop('color')};
  border-radius: 4px 0 0 4px;
  padding: 8px;
  min-width: 40px;
  box-sizing: border-box;
`

export const TextWrapper = sc.div`
  border: 1px solid ${theme('greyMid')};
  border-left: none;
  /* Account for border */
  padding: 8px 7px 8px 8px;
  border-radius: 0 4px 4px 0;
  ${text4}
  font-weight: 500;
`
