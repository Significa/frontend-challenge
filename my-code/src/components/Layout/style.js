import sc from 'styled-components'

// eslint-disable-next-line import/prefer-default-export
export const Outer = sc.div`
  display: flex;
  flex-direction: column;
  max-width: 1228px;
  margin: 0 auto;
  padding: 24px;
  min-height: 100vh;
  box-sizing: border-box;
`
