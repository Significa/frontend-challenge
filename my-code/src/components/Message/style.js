import sc from 'styled-components'
import { theme } from 'styled-tools'

import { text2, text4 } from '../../styles/text'

import illustration from '../../assets/illustration-empty-state@2x.png'

export const Center = sc.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`

export const Illustration = sc.img.attrs({
  src: illustration,
  alt: ''
})`
  object-fit: contain;
  width: 400px;
  max-width: 100%;
  height: 200px;
`

export const Title = sc.span`
  ${text2}
  margin-top: 16px;
`

export const Subtitle = sc.span`
  ${text4}
  color: ${theme('greyLight')};
  margin-top: 8px;
`
