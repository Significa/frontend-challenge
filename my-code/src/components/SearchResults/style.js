import sc from 'styled-components'

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = sc.div`
  margin-top: 8px;
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(auto-fill, minmax(155px, 1fr));

  @media (max-width: 800px) {
    grid-template-columns: repeat(auto-fill, minmax(140px, 1fr));
  }

  @media (max-width: 400px) {
    grid-template-columns: repeat(auto-fill, minmax(125px, 1fr));
  }
`
