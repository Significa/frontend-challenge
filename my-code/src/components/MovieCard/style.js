import sc from 'styled-components'
import { Link } from 'react-router-dom'
import { theme } from 'styled-tools'

import { text3, text4 } from '../../styles/text'
import IconButton from '../IconButton'

export const Wrapper = sc.div`
  width: 100%;
  text-decoration: none;
  position: relative;
  border-radius: 3px;
  overflow: hidden;
  color: ${theme('greyWhite')};

  &:focus {
    outline: none;
  }

  /* Aspect ratio */
  &:before {
    content: '';
    display: block;
    padding-top: 133.3333333%;
  }
`

export const CoverLink = sc(Link)`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

export const Poster = sc.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: ${theme('greyMain')};
  background-image: ${({ poster }) => `url(${poster})`};
`

export const FavoriteButtonWrapper = sc(IconButton)`
  height: auto;
  position: absolute;
  z-index: 1;
  top: 12px;
  right: 12px;
  color: ${theme('greyWhite')};
  will-change: opacity;
  transition:
    opacity
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;
  filter: drop-shadow(0 0 2px rgba(0, 0, 0, .2));
  opacity: ${({ visible = true }) => (visible ? 1 : 0)};

  ${Poster} ~ ${CoverLink}:hover ~ &,
  ${Poster} ~ ${CoverLink}:focus ~ &,
  ${Poster} ~ &:hover,
  ${Poster} ~ &:focus {
    opacity: 1;
  }
`

export const Title = sc.span`
  ${text3}
`

export const Subtitle = sc.span`
  ${text4}
`

export const Content = sc.div`
  pointer-events: none;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  padding: 12px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background: ${theme('scrimColor')};
  will-change: opacity;
  transition:
    opacity
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;

  ${Poster} ~ & {
    opacity: 0;
  }

  ${Poster} ~ ${CoverLink}:hover ~ &,
  ${Poster} ~ ${CoverLink}:focus ~ &,
  ${Poster} ~ ${FavoriteButtonWrapper}:hover ~ &,
  ${Poster} ~ ${FavoriteButtonWrapper}:focus ~ & {
    opacity: 1;
  }
`
