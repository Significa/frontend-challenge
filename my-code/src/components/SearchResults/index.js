import React from 'react'
import PropTypes from 'prop-types'

import MovieCard from '../../containers/MovieCard'

import { Wrapper } from './style'

export default function SearchResults({ ids }) {
  return (
    <Wrapper>
      {ids.map(id => (
        <MovieCard key={id} id={id} />
      ))}
    </Wrapper>
  )
}

SearchResults.propTypes = {
  ids: PropTypes.arrayOf(PropTypes.string).isRequired
}
