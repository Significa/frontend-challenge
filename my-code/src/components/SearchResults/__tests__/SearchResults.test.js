import React from 'react'
import { shallow } from 'enzyme'

import SearchResults from '../'
import { Wrapper } from '../style'

describe('SearchResults', () => {
  it('renders correctly', () => {
    const tree = shallow(<SearchResults ids={['tt0086567', 'tt0434409']} />)
    expect(tree).toMatchSnapshot()
  })

  describe('Wrapper', () => {
    it('renders correctly', () => {
      const tree = shallow(<Wrapper />)
      expect(tree).toMatchSnapshot()
    })
  })
})
