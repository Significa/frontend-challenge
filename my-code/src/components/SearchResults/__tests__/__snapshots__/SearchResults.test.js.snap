// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Wrapper renders correctly 1`] = `
.c0 {
  margin-top: 8px;
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(auto-fill,minmax(155px,1fr));
}

@media (max-width:800px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,minmax(140px,1fr));
  }
}

@media (max-width:400px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,minmax(125px,1fr));
  }
}

<div
  className="c0"
/>
`;

exports[`SearchResults renders correctly 1`] = `
<Wrapper>
  <Connect(MovieCard)
    id="tt0086567"
    key="tt0086567"
  />
  <Connect(MovieCard)
    id="tt0434409"
    key="tt0434409"
  />
</Wrapper>
`;
